skimage/filters/rank/generic.py

```diff
@@ -385,11 +385,13 @@ def autolevel(image, footprint, out=None, mask=None,
         return _apply_scalar_per_pixel(generic_cy._autolevel, image, footprint,
                                        out=out, mask=mask,
                                        shift_x=shift_x, shift_y=shift_y)
-    else:
+    if np_image.ndim == 3:
```

Comment:
Nitpick: I think the current logic is a bit unexpected. It's a bit tedious but I think the form
```python
if np_image.ndim == 2:
    ...
elif np_image.ndim == 3:
    ...
else:
    ... 
```
would be a bit easier to maintain. I'm totally happy to do it myself but wanted to check in with you first to avoid merge conflicts with your local version. May I go ahead? :wink:

Reply:
The linter suggested it because each condition returns, so if a condition passed we won't get to the next one anyway.
If you think we should keep the `else`s I can change it back :)
(I have some changes to do anyway)

Reply:
Yeah, the logic is totally valid! I guess it's just my bias being at work here about what I seems easier to read to me. If I see an `elif` I know that there is another `else` condition immediately. 

In some cases where there is a single return statement at the end of the function, there may be subtle bugs, e.g. when both conditions are not mutually exclusive. But unlikely here with the early `return`.

Forgive my rambling, I'm fine with keeping it as it is. :wink:

Reply:
I'm with @lagru, fine to leave this as is, but I agree that the linter is wrong here. Linters care about logic flow, but Python started as a teaching language. From an English comprehension perspective, the if/elif/else form is so much easier than what the linter suggests.